backend/pkg/api/connect/service/acl/service.go

```diff
@@ -156,10 +156,52 @@ func (s *Service) CreateACL(ctx context.Context, req *connect.Request[v1alpha1.C
 }
 
 // DeleteACLs implements the handler for the delete ACL endpoint.
-func (*Service) DeleteACLs(context.Context, *connect.Request[v1alpha1.DeleteACLsRequest]) (*connect.Response[v1alpha1.DeleteACLsResponse], error) {
-	return nil, apierrors.NewConnectError(
-		connect.CodeUnimplemented,
-		errors.New("endpoint is not implemented"),
-		apierrors.NewErrorInfo(v1alpha1.Reason_REASON_CONSOLE_ERROR.String()),
-	)
+func (s *Service) DeleteACLs(ctx context.Context, req *connect.Request[v1alpha1.DeleteACLsRequest]) (*connect.Response[v1alpha1.DeleteACLsResponse], error) {
+	// TODO: Ensure that neither req, req.Msg or req.Msg.Filter can never be nil
+	kafkaReq, err := s.kafkaClientMapper.deleteACLFilterToDeleteACLKafka(req.Msg.Filter)
+	if err != nil {
+		return nil, apierrors.NewConnectError(
+			connect.CodeInternal, // Internal because all input should already be validated, and thus no err possible
+			err,
+			apierrors.NewErrorInfo(v1alpha1.Reason_REASON_CONSOLE_ERROR.String()),
+		)
+	}
+
+	res, err := s.consoleSvc.DeleteACLsKafka(ctx, kafkaReq)
+	if err != nil {
+		return nil, apierrors.NewConnectError(
+			connect.CodeInternal,
+			err,
+			apierrors.NewErrorInfo(v1alpha1.Reason_REASON_KAFKA_API_ERROR.String(), apierrors.KeyValsFromKafkaError(err)...),
+		)
+	}
+
+	if len(res.Results) != 1 {
```

Comment:
quesiton: if this is always an operation over a single entity, shouldn't it be called `DeleteACL`?

Reply:
This is a good question and I'm not entirely sure. So with one such filter you can delete multiple ACLs actually (any ACLs that match the filter), and in fact you could construct one filter that matches all ACLs by using the rigth enums (ANY operation, ANY resource pattern type, etc). However we could offer a second endpoint that accepts multiple filters in one request, since this is what the Kafka API does under the hood as well.

@twmb @birdayz What are your opinions on this?

Reply:
I'll proceed with DeleteACLs because it deletes multiple ACLs, but we could add another DeleteACLsBulk or something in the future which accepts more than one filter. The response handling is more complex (due to partial errors/success) though, so both methods would have their right to exist I believe